core/shared/src/main/scala/zio/IO.scala

```diff
@@ -523,6 +523,11 @@ object IO {
    */
   def fromOption[A](v: => Option[A]): IO[Option[Nothing], A] = ZIO.fromOption(v)
 
+  /**
+   * @see See [[zio.ZIO.getOrFailUnit]]
+   */
+  def getOrFailUnit[A](v: => Option[A]): IO[Unit, A] = ZIO.getOrFailUnit(v)
```

Comment:
@adamgfraser  Shouldn't this be `IO[None.type, A]` as per more recent convention?

Reply:
@jdegoes The goal of this constructor was to have a variant for users who want `Unit` in the error channel instead of `None.type`. So we have `ZIO.fromOption` for the normal recommended usage, `ZIO.getOrFailUnit` for if you really want `Unit` in the error channel, and `ZIO.getOrFail` for if you want `NoSuchElementException` in the error channel (for example, for unifying with other errors that are a subtype of `Throwable`). Does that make sense?